source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -482,11 +482,169 @@ account:
         let amount1 = Amount::from_ccd(20);
         claim_eq!(host.get_transfers_to(receiver0), [amount0, amount1]);
 
+.. _writing_property_based_tests:
+
+Writing property-based tests
+============================
+
+Property-based testing technique allows for testing statements about your code that expected to be true for any input parameters.
+The input to such tests is generated randomly.
+Property-based testing is supported using the |QuickCheck|_ crate.
+The tests should placed in the same module as Wasm unit tests and annotated with the ``#[concordium_quickcheck]`` macro.
+The macro takes a named attribute ``num_tests`` for specifying the number of random tests to run.
```

Comment:
Consider moving this line to the example, as that gives more context for the user to understand it. The user might not know what a named attribute is in a macro. But seeing it with the example makes it quite obvious.